src/main/java/hudson/plugin/versioncolumn/JVMVersionComparator.java

```diff
@@ -92,8 +92,15 @@ private int getMasterBytecodeMajorVersionNumber() {
     private boolean isAgentRuntimeCompatibleWithJenkinsBytecodeLevel(String agentMajorMinorVersion) {
         Integer masterBytecodeLevel = getMasterBytecodeMajorVersionNumber();
         Integer agentVMMaxBytecodeLevel = JDK_VERSION_NUMBER_TO_BYTECODE_LEVEL_MAPPING.get(agentMajorMinorVersion);
-
-        return masterBytecodeLevel <= agentVMMaxBytecodeLevel;
+        if (agentVMMaxBytecodeLevel != null) {
+            return masterBytecodeLevel <= agentVMMaxBytecodeLevel;
+        } else {
+            LOGGER.log(Level.WARNING, Messages.JVMVersionMonitor_UnrecognizedAgentJVM(agentMajorMinorVersion));
+            /*
+             * Even if the version might be compatible, we still mark the node as incompatible to prevent potential issues.
+             */
+            return false;
```

Comment:
This looks fine.
I agree with Oleg it would be nice to improve, but I think Adrien is also right that the immediate need is to fix the NPE and re-release ASAP so the plugin is de``FUBAR``ed.

NIT: 3 lines for that comment instead of using `//` hurts slightly :P.